### PR TITLE
Add sourceforge (git) backend to retrieve git tags

### LINKED ISSUE #223: Add support for (SourceForge) git tags
Some projects use git tags in public git repositories to mark releases. Support for those would be helpful.

### PR DESCRIPTION
**Pull request by LenkaSeg**
New module handling the extraction of git tags from Sourceforge.net

Closes #223

### CODE REVIEW

**Comment by LenkaSeg** [comment]
Oops, forgot to write tests for it :)

File: docs/user-guide.rst [hunk 1]
```diff
@@ -207,6 +207,9 @@ The backends available are:
   You need to provide **Sourceforge name** if the name on RSS feed is different then the
   project name on Sourceforge.
 
+* **Sourceforge (git) for projects hosted on
```
Comment (Zlopez) [inline docs/user-guide.rst:210]: It will be good to describe the difference between this backend and the sourceforge one.

**Review by Zlopez — CHANGES_REQUESTED** [review]
Just one thing in documentation

File: docs/user-guide.rst (hunk 1 above)
Comment (LenkaSeg) [reply docs/user-guide.rst:210]: Ok! Will do that.

File: anitya/lib/backends/sourceforge_git.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        url = project.get_version_url()
+        git_tag_request = requests.get(url)
+
+        if git_tag_request.status_code == 404:
+            raise AnityaPluginException(
+                'Could not call : "%s" of "%s", with error: %s'
+                % (url, project.name, git_tag_request.status_code)
+            )
+
+        soup = BeautifulSoup(git_tag_request.content, "html.parser")
```
Comment (Zlopez) [inline anitya/lib/backends/sourceforge_git.py:121]: The approach with BeatifulSoup is really nice. I wonder why it's not used in some other backends.

File: anitya/tests/lib/backends/test_sourceforge_git.py [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# Any Red Hat trademarks that are incorporated in the source
+# code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission
+# of Red Hat, Inc.
+#
+
+"""
+anitya tests for the custom backend.
```
Comment (Zlopez) [inline anitya/tests/lib/backends/test_sourceforge_git.py:23]: This should be updated

File: anitya/tests/lib/backends/test_sourceforge_git.py [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+from anitya.db import models
+from anitya.lib.exceptions import AnityaPluginException
+from anitya.tests.base import DatabaseTestCase, create_distro
+
+
+BACKEND = "Sourceforge (git)"
+
+
+class SourceforgeGitBackendtests(DatabaseTestCase):
+    """Sourceforge backend tests."""
```
Comment (Zlopez) [inline anitya/tests/lib/backends/test_sourceforge_git.py:38]: This should be updated as well.

**Review by Zlopez — CHANGES_REQUESTED** [review]
Looks very good, just found a few small typos in comments.

File: anitya/lib/backends/sourceforge_git.py [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+import requests
+
+from anitya.lib.backends import BaseBackend
+from anitya.lib.exceptions import AnityaPluginException
+
+
+class SourceforgeGitBackend(BaseBackend):
+    """The custom class for projects hosted on sourceforge.net.
+
+    This backend allows to specify a version_url and a regex that will
```
Comment (Zlopez) [inline anitya/lib/backends/sourceforge_git.py:13]: I don't think we are using regex for this backend.

**Comment by LenkaSeg** [comment]
I hope I fixed all the comments.
I noticed there is one condition missing in `backends/sourceforge_git:get_namespace_repo`, so I added that too.

**Comment by Zlopez** [comment]
Just fix the failing diff-cover and we can merge this.

**Review by Zlopez — APPROVED** [review]

**Pull request merged**